app/components/channel_drawer/channels_list/channel_item/index.js

```diff
@@ -18,8 +19,15 @@ function makeMapStateToProps() {
             member = getMyChannelMember(state, ownProps.channelId);
         }
 
+        let teammate = null;
+        if (channel.teammate_id) {
+            teammate = getUser(state, channel.teammate_id);
```

Comment:
Why are we getting the user if we only need the id which we already have?

more on that don't you think instead of passing the `currentUserId` and `channelTeammateId` as props you can pass something like `isMyUser` that is equals to `currentUserId === channel.teammate_id`?

Reply:
one other thing I'm seeing **channel.teammate_id** and we definitely shouldn't be using that as teammate_id is not a property of the channel and is being injected by a selector somewhere that we should remove.

Perhaps check if the channel.type is a DM and then get the teammate_id from it.

Reply:
Agree on your points and teammate is not necessary at this stage. But this will come handy when we implement the "DM channel with deactivated users" feature similar to webapp https://mattermost.atlassian.net/browse/PLT-7917.
I'll update the change as you commented above. Thanks!